sample-app/src/components/InputDropdown.tsx

```diff
@@ -12,7 +12,7 @@ interface Props {
   optionIdPrefix: string,
   onSubmit?: (value: string) => void,
   updateInputValue: (value: string) => void,
-  updateDropdown: () => void,
+  updateDropdown: () => Promise<void>,
```

Comment:
Was there a problem that required making this async?

Reply:
yeah, since `onClick` calls updateDropdown and dispatch for 'showOptions', without async wait for updateDropdown, autocomplete would flicker between previous result shown and the latest result during render (i.e. showOptions dispatch will trigger display for the old results before updateDropdown finish executing a new search)

Reply:
With this approach, some callbacks are async and other are not, and that distinction might seem arbitrary to a consumer of this component. I think the simplest solution would be to update the code inside the `onClick` handler to use the `.then` syntax (e.g. `updateDropdown().then(...` so that we can avoid making this async. Alternatively we might want to consider supporting both async and non-async callbacks (I'd want to have a larger discussion with Tom before going this route)

Reply:
hm I see. updateDropdown would have to return Promise<void> in order to use the `.then` syntax. Perhaps we can make a distinction with the naming between the callbacks that requires network calls and those that's just state modification like tracking input string? I will talk to tom if we want to support non-async callbacks as well

Reply:
would it make sense to move this to a different PR?

Reply:
actually ignore me it's probably best to just resolve this here

Reply:
removed the async stuff for now! will add support for non-async and async callbacks in another item